code/modules/fish/fish_tank.dm

```diff
@@ -0,0 +1,682 @@
+	icon_state = "tank1"
+	density = TRUE
+	anchored = TRUE
+	throwpass = TRUE				// You can throw objects over this, despite it's density, because it's short enough.
+
+	tank_type = "tank"
+	water_capacity = 200		// Decent sized, holds 2 full large beakers worth
+	max_fish = 4				// Room for a few fish
+
+	has_lid = TRUE
+	max_health = 50				// Average strength, will take a couple hits from a toolbox.
+	cur_health = 50
+	shard_count = 2
+
+
+/obj/machinery/fishtank/wall
+	name = "wall aquarium"
+	desc = "This aquarium is massive! It completely occupies the same space as a wall, and looks very sturdy too!"
+	icon_state = "wall1"
+	density = TRUE
+	anchored = TRUE
+	throwpass = FALSE				// This thing is the size of a wall, you can't throw past it.
+
+	tank_type = "wall"
+	water_capacity = 500		// This thing fills an entire tile,5 large beakers worth
+	max_fish = 10				// Plenty of room for a lot of fish
+
+	has_lid = TRUE
+	max_health = 100			// This thing is a freaking wall, it can handle abuse.
+	cur_health = 100
+	shard_count = 3
+
+/obj/machinery/fishtank/wall/Cross(atom/movable/mover, turf/target, height = 1.5, air_group = 0) // Prevents airflow. Copied from windows.
+	if(istype(mover) && mover.checkpass(PASSGLASS))
+		return TRUE
+	return FALSE
+
+/obj/machinery/fishtank/wall/Uncross(atom/movable/O as mob|obj, target as turf)
+	return TRUE
+
+//////////////////////////////
+//		VERBS & PROCS		//
+//////////////////////////////
+
+/obj/machinery/fishtank/verb/toggle_lid_verb()
+	set name = "Toggle Tank Lid"
+	set category = "Object"
+	set src in view(1)
+	if (usr.incapacitated() || usr.isDead() || !usr.Adjacent(src))
+		return FALSE
+	toggle_lid(usr)
+
+/obj/machinery/fishtank/proc/toggle_lid(var/mob/living/user)
+	lid_switch = !lid_switch
+	update_icon()
+
+/obj/machinery/fishtank/verb/toggle_light_verb()
+	set name = "Toggle Tank Light"
+	set category = "Object"
+	set src in view(1)
+	if (usr.incapacitated() || usr.isDead() || !usr.Adjacent(src))
+		return FALSE
+	toggle_light(usr)
+
+/obj/machinery/fishtank/proc/toggle_light(var/mob/living/user)
+	light_switch = !light_switch
+	if(light_switch)
+		set_light(2,2,"#a0a080")
+	else
+		set_light(0)
+
+//////////////////////////////
+//		NEW() PROCS			//
+//////////////////////////////
+
+/obj/machinery/fishtank/New()
+	..()
+	if(!has_lid)				//Tank doesn't have a lid/light, remove the verbs for then
+		verbs -= /obj/machinery/fishtank/verb/toggle_lid_verb
+		verbs -= /obj/machinery/fishtank/verb/toggle_light_verb
+
+/obj/machinery/fishtank/tank/New()
+	..()
+	if(prob(5))					//5% chance to get the castle decoration
+		icon_state = "tank2"
+
+//////////////////////////////
+//		ICON PROCS			//
+//////////////////////////////
+
+/obj/machinery/fishtank/update_icon()
+	overlays.Cut()
+
+	//Update Alert Lights
+	if(has_lid)											//Skip the alert lights for aquariums that don't have lids (fishbowls)
+		if(egg_list.len > 0)								//There is at least 1 egg to harvest
+			overlays += "over_egg"
```

Comment:
It seems crazy to me to use overlays to show everything EXCEPT fish

It's a fish tank, you wanna see the fish inside, right?

Reply:
Most of these overlays are just for the lights
There is currently no sprite of the fish inside the tanks, and I'm afraid I'm not able to do it properly :(